### PR TITLE
Adds support macOS dylib

### PR DESCRIPTION
**Pull request by pftg**

### CODE REVIEW

File: ext/extconf.rb [hunk 1]
```diff
@@ -83,10 +83,9 @@ def safe_sh(cmd)
 
 # Absolutely prevent the linker from picking up any other zookeeper_mt
 Dir.chdir("#{HERE}/lib") do
-  %w[st mt].each do |stmt|
```
Comment (eric) [inline ext/extconf.rb]: Why remove the line copying the `mt` version?

Comment (pftg) [reply ext/extconf.rb]: based on https://github.com/zk-ruby/zookeeper/pull/86/files#diff-b3c266c0a40fd6ea5ce51093b147dbe6R93 there is no `mt` usage. I could split into 2 commits with dylib adding and then with small refactoring

Comment (pftg) [reply ext/extconf.rb]: I think, yep I'll create separate PR for that refactoring

**Comment by pftg** [comment]
Hi @eric please check again. I removed extra cleanups. Just added support for `.dylib`

File: ext/extconf.rb (hunk 1 above)
Comment (pftg) [reply ext/extconf.rb]: Updated PR without this change

**Comment by pftg** [comment]
@eric @slyphon hi guys, what do you think about this PR?

**Pull request merged**